custom_components/octopus_intelligent/persistent_data.py

```diff
@@ -0,0 +1,93 @@
+"""Persistent data storage for the integration, based on the HASS helpers.storage.Store class."""
```

Comment:
The `PersistentData` class introduced here is fairly generic and could be used to save any other data when/if needed. Building on the Home Assistant `helpers.storage.Store` class, it will create / load / save / delete a JSON file in the HASS `config/.storage/` directory. For example, in a HASS container:

```
edea03bac07a:~# cd /config/.storage

edea03bac07a:/config/.storage# ls -l
-rw-------    1 root     root           373 Mar  3 16:52 core.config
-rw-r--r--    1 root     root          2826 Mar  5 00:39 core.config_entries
...
-rw-r--r--    1 root     root           156 Mar  5 02:21 octopus_intelligent.A-1234ABCD

edea03bac07a:/config/.storage# cat octopus_intelligent.A-1234ABCD
{
  "version": 1,
  "minor_version": 1,
  "key": "octopus_intelligent.A-1234ABCD",
  "data": {
    "last_seen_planned_dispatch_source": "smart-charge"
  }
```
Where `A-1234ABCD` is the user’s Octopus account number.

I have also taken care of adding the `async_remove_entry()` function to `__init__.py` in order to arrange for the store file (`'octopus_intelligent.A-1234ABCD'`) to be _deleted_ if the user deletes the integration using the HASS frontend web UI.